Urban behaviors rework

Badulundabad opened the pull request:
- reworked assimilation behavior and component. Now it doesn't turn empire men into vampires, notables in all settlements are humans;
- removed excess extensions of settlement and hero classes;
- changed wanderer behavior to avoid cases where vampire companion is still sitting in empire town and vice versa.

Badulundabad committed:
changed assimilation behavior and component

Badulundabad committed:
removed excess hero extensions

Badulundabad committed:
removed excess settlement extensions

Badulundabad committed:
changed wanderers behavior

Badulundabad committed:
Merge branch 'development' into urban_behaviors_rework

hunharibo commented on CSharpSourceCode/CampaignSupport/Assimilation/AssimilationCampaignBehavior.cs:
```diff
@@ -13,18 +11,26 @@ public class AssimilationCampaignBehavior : CampaignBehaviorBase
         public override void RegisterEvents()
         {
             CampaignEvents.OnGameLoadFinishedEvent.AddNonSerializedListener(this, OnGameLoadFinishedEvent);
```
I would suggest to use the OnSessionLaunched event instead. I think GameLoadFinished only fires when loading a savegame, not when starting a new game, and my guess is that you want both.

Badulundabad replied:
I don't add this component to every settlement at new game start, but to each settlement after its owner is changed and only if culture is different. After loading a save game I need to add already created components to their settlements. That's why I use OnGameLoadFinishedEvent

hunharibo replied:
OK, good enough :)

hunharibo commented on CSharpSourceCode/CampaignSupport/Assimilation/AssimilationComponent.cs:
```diff
@@ -1,284 +1,100 @@
             {
-                Campaign.Current.IssueManager.ChangeIssueOwner(victim.Issue, newNotable);
+                var outrider = outriders.First();
+                outrider.Culture = _settlement.Owner.Culture;
```
I am not sure that setting the notable's culture directly changes the troops recruitable by them. Have you tested? If yes, that is very good news! I thought we would have to create new notables based on the xml settings in culture for it to work.,

Badulundabad replied:
I've tested it a few months ago and it worked like a swiss watch

hunharibo replied:
Saw your picture as well on discord. Fine then, no more issues.

hunharibo reviewed (approved):
Approved after resolving the 2 comments. Nice work!

Z3rca reviewed (approved):
👍

Badulundabad committed:
fixed AssimilationComponent and WanderersCampaignBehavior

- now AssimilationComponent fires AssimilationComplete event when it's needed;
- wanderers behavior checks only towns;
